src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,88 @@
+package duke;
+
+import duke.task.TaskList;
+import duke.task.Todo;
+import duke.task.Deadline;
+import duke.task.Event;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+    private FileManager fileManager;
+    private TaskList taskList;
+
+    public Parser(FileManager fileManager, TaskList taskList) {
+        this.fileManager = fileManager;
+        this.taskList = taskList;
+    }
+    public void parse(String userMessage) {
```

Comment:
This method might be abit too long. Maybe you can abstract each of the code in each block? [Avoid long methods](https://nus-cs2103-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html)